pkg/validation/validation.go

```diff
@@ -0,0 +1,51 @@
+package validation
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const ErrorNotFoundAgent = "unsupported fence agent %s:"
+
+type validateAgentExistence struct {
+	isFileExist func(string) (fs.FileInfo, error)
+}
+
+type AgentValidator interface {
+	ValidateAgentName(agent string) (admission.Warnings, error)
+}
+
+func NewAgentValidator(fileGetter func(string) (fs.FileInfo, error)) AgentValidator {
+	return &validateAgentExistence{isFileExist: fileGetter}
+}
+
+func (vfe *validateAgentExistence) ValidateAgentName(agent string) (admission.Warnings, error) {
+	exist, err := vfe.isAgentSupported(agent)
```

Comment:
- what's the value of calling another function?
- error handling is... strange? Probably because exists is always false when err != nil? That's confusing.